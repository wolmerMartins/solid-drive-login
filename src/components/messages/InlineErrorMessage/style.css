.InlineErrorMessage {
  margin-top: 1rem;
  color: var(--red-main);
  font-size: var(--xxs-font);
}
