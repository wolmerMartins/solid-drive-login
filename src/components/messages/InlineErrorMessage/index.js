import React from 'react'

import './style.css'

const InlineErrorMessage = ({ message = '' }) => (
  <div className="InlineErrorMessage">
    {message}
  </div>
)

export default InlineErrorMessage
