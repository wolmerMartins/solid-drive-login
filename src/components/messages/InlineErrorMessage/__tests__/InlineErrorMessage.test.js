import React from 'react'
import { render } from '@testing-library/react'

import InlineErrorMessage from '../index'

const buildComponent = props => {
  const utils = render(<InlineErrorMessage {...props} />)
  const component = props.message
    ? utils.getByText(props.message)
    : utils.asFragment()

  return {
    ...utils,
    component
  }
}

describe('<InlineErrorMessage />', () => {
  it('Render the component with an empty message', () => {
    const { component } = buildComponent({})

    expect(component)
      .toMatchSnapshot()
  })

  it('Render the component with \'error inline component\' message', () => {
    const { component } = buildComponent({ message: 'error inline component' })

    expect(component)
      .toMatchSnapshot()

    expect(component.textContent)
      .toBe('error inline component')
  })
})
