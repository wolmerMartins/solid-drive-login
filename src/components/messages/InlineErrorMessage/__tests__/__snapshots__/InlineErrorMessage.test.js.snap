// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineErrorMessage /> Render the component with 'error inline component' message 1`] = `
<div
  class="InlineErrorMessage"
>
  error inline component
</div>
`;

exports[`<InlineErrorMessage /> Render the component with an empty message 1`] = `
<DocumentFragment>
  <div
    class="InlineErrorMessage"
  />
</DocumentFragment>
`;
